src/app/services/filesystem.service.ts

```diff
@@ -20,28 +20,28 @@ export interface ProviderOptions {
   datasetsAndZvols?: boolean;
   zvolsOnly?: boolean;
   datasetsOnly?: boolean;
+  shouldDisableNode?: (node: ExplorerNodeData) => Observable<boolean>;
 }
 
-const roolZvolNode = {
-  path: zvolPath,
-  name: zvolPath,
-  hasChildren: true,
-  type: ExplorerNodeType.Directory,
-} as ExplorerNodeData;
-
-const roolDatasetNode = {
-  path: '/mnt',
-  name: '/mnt',
-  hasChildren: true,
-  type: ExplorerNodeType.Directory,
-};
-
 @Injectable({ providedIn: 'root' })
 export class FilesystemService {
   constructor(
     private api: ApiService,
   ) {}
 
+  getTopLevelDatasetsNodes(providerOptions: ProviderOptions): Observable<ExplorerNodeData[]> {
```

Comment:
If this method is only used to retrieve top level datasets, then most of the options here are not applicable.

Reply:
What is this method supposed to return?

Reply:
How could it be something different? The top level nodes for the filesystem would always be root datasets, no? Never zvols, never snapshots, never directories or hidden files, so I don't see now options can change anything.